src/mlpack/methods/ann/layer/padding_impl.hpp

```diff
@@ -42,10 +46,32 @@ void Padding<InputDataType, OutputDataType>::Forward(
 {
   nRows = input.n_rows;
   nCols = input.n_cols;
-  output = arma::zeros(nRows + padWLeft + padWRight,
-      nCols + padHTop + padHBottom);
-  output.submat(padWLeft, padHTop, padWLeft + nRows - 1,
-      padHTop + nCols - 1) = input;
+  
+  if (inputWidth == 0 || inputHeight == 0)
+  {
+    output = arma::zeros(nRows + padWLeft + padWRight,
+        nCols + padHTop + padHBottom);
+    output.submat(padWLeft, padHTop, padWLeft + nRows - 1,
+        padHTop + nCols - 1) = input;
+  }
+  else
+  {
```

Comment:
Might be cleaner to remove else statement simply set output values here and return.

Reply:
I can do that but that would just mean writing the same `outputWidth` and `outputHeight` code two times. Is this what you meant ?

Reply:
held this because wasn't sure about this.

Reply:
Not sure either, I guess @kartikdutt18 likes to add a `return` statement after line 55, and just remove the else case? On the other side, I think we will remove the first case anyway, right now it's just a quick patch so we don't have to update the existing layers/tests?

Reply:
for the current implementation we would need the two cases because of the input dimensions if they are zero there will be a float div error

Reply:
and i just believe it's a better idea to keep two cases so no need to change other impls because even now i face a 0 error with max pool layer in my model, though a different case.

Reply:
We can use C++11 features right, so C++ has defer function, so you can right the statement once [set outputWidth and outputHeight]. It will be a lot cleaner.

Reply:
Testing if we can use this.

Reply:
I have never used defer so let me take a look at it and if i can i will chage the code that way.

Reply:
Will merge this once builds come back green, Will raise a PR for the changes in this and other layers that call Padding layer.

Reply:
Sounds good i just need to fix the tests one more time :D